admin-writer/src/main/kotlin/com/epam/drill/admin/writer/rawdata/route/RawDataWriterRoutes.kt

```diff
@@ -126,4 +133,34 @@ internal suspend fun sendPostRequest(url: String, data: Any) {
         contentType(ContentType.Application.Json)
         body = data
     }
+}
+
+internal val json = Json {
+    ignoreUnknownKeys = true
+    explicitNulls = false
+}
+
+/**
+ * Workaround for decompressing the request body before upgrading to Ktor 3.0.0, where this feature works out of the box
+ * https://github.com/ktorio/ktor/issues/3845
+ */
+internal suspend inline fun <reified T : Any> ApplicationCall.decompressAndReceive(): T {
+    val body: ByteArray = when (request.headers[HttpHeaders.ContentEncoding]) {
+        "gzip" -> decompressGZip(receiveStream())
+        else -> receive<ByteArray>()
+    }
+    return when (request.headers[HttpHeaders.ContentType]) {
+        ContentType.Application.ProtoBuf.toString() -> ProtoBuf.decodeFromByteArray(T::class.serializer(), body)
+        ContentType.Application.Json.toString() -> json.decodeFromString(T::class.serializer(), String(body))
+        else -> throw UnsupportedMediaTypeException(
+            ContentType.parse(request.headers[HttpHeaders.ContentType] ?: "application/octet-stream")
```

Comment:
Defaulting to `?: "application/octet-stream"` is misleading. I suggest replacing it with `?: "Content-Type header is missing"`